extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java

```diff
@@ -174,7 +174,7 @@ public Long getPosition(StreamPartition<Integer> partition)
   }
 
   @Override
-  public void close()
+  public synchronized void close()
```

Comment:
Please add a Javadoc explaining why it's synchronized.

Reply:
Done

Reply:
Usually Kafka consumer is closed via `wakeup` method which is thread-safe and can be called concurrently with the other methods in KafkaConsumer. If one thread is busy in `poll` and another thread tries to `close`, it will likely result in an exception.